### PR TITLE
Added CLUSTER INFO support when cluster enabled

### PR DESCRIPTION
**Pull request by vvanholl**

### CODE REVIEW

File: exporter/redis.go [hunk 1]
```diff
@@ -436,16 +436,31 @@ func (e *Exporter) scrape(scrapes chan<- scrapeResult) {
...
 		log.Debugf("connected to: %s", addr)
 
+		is_cluster := false
+
 		info, err := redis.String(c.Do("INFO", "ALL"))
 		if err == nil {
 			err = e.extractInfoMetrics(info, addr, scrapes)
-		}
-		if err != nil {
+			if strings.Index(info, "cluster_enabled:1") != -1 {
```
Comment (oliver006) [inline exporter/redis.go]: Why not move that `if` check to line 453 and only do the `CLUSTER INFO` if the first `extractInfoMetrics` in 443 succeeded? That way you also won't need the extra `is_cluster` variable.

File: exporter/redis.go [hunk 2]
```diff
@@ -436,16 +436,31 @@ func (e *Exporter) scrape(scrapes chan<- scrapeResult) {
...
+		} else {
 			log.Printf("redis err: %s", err)
 			errorCount++
 			continue
 		}
 
+		if is_cluster {
+			info, err = redis.String(c.Do("CLUSTER", "INFO"))
+			if err == nil {
+				err = e.extractInfoMetrics(info, addr, scrapes)
```
Comment (oliver006) [inline exporter/redis.go]: Any more metrics that need to be added to `metricMap` as a result of this?

**Comment by oliver006** [comment]
Thanks for the PR!

File: exporter/redis.go (hunk 1 above)
Comment (vvanholl) [inline exporter/redis.go]: You are right. I am updating code for a new commit.

File: exporter/redis.go (hunk 2 above)
Comment (vvanholl) [inline exporter/redis.go]: metricMap role is to give more fancy names to metrics ?

Comment (oliver006) [inline exporter/redis.go]: Yeah, see #33 which Brian submitted to improve metrics names in line with Prometheus best practices.

Comment (oliver006) [inline exporter/redis.go]: I see you added two names to the map - let me know if this are all the names you want to add and I'll merge this.

**Comment by vvanholl** [comment]
@oliver006 According to my opinion,  these are the 2 only metrics which should change name.

**Comment by oliver006** [comment]
👍

**Pull request merged**